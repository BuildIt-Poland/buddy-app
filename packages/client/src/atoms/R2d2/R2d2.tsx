import React from 'react';
import { ReactComponent as Robot } from 'assets/svg/r2d2.svg';
import makeStyles from '@material-ui/core/styles/makeStyles';

const useStyles = makeStyles(theme => ({
  r2d2: {
    animation: `$float 3s ${theme.transitions.easing.easeInOut} infinite`,
  },
  '@keyframes float': {
    '0%': {
      transform: 'translateY(1rem)',
    },
    '50%': {
      transform: 'translateY(-1rem)',
    },
    '100%': {
      transform: 'translateY(1rem)',
    },
  },
}));

const R2d2: React.FC = () => {
  const classes = useStyles();

  return <Robot className={classes.r2d2} />;
};

export default R2d2;
