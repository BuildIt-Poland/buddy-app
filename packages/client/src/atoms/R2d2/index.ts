import R2d2 from './R2d2';

export default R2d2;
