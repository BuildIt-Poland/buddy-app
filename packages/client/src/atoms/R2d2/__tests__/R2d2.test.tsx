import React from 'react';
import { create } from 'react-test-renderer';

import R2d2 from '../R2d2';

describe('Component - R2d2', () => {
  test('renders correctly', () => {
    const component = create(<R2d2 />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
