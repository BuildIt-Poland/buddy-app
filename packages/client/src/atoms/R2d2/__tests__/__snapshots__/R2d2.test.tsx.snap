// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - R2d2 renders correctly 1`] = `
<svg
  className="makeStyles-r2d2-1"
>
  r2d2.svg
</svg>
`;
